CSharp/Library/Microsoft.Bot.Connector.Shared/MicrosoftAppCredentials.cs

```diff
@@ -266,6 +284,13 @@ private bool TokenNotExpired(OAuthResponse token)
             return token.expiration_time > DateTime.UtcNow;
         }
 
+        private bool TokenWithinSafeTimeLimits(OAuthResponse token)
+        {
+            int secondsToHalfwayExpire = Math.Min(token.expires_in / 2, 1800);

```

Comment:
>token.expires_in [](start = 50, length = 16)

would it be more consistent to use TimeToExpiration.TotalSeconds here?

Reply:
expires_in is the field coming back in the OAuth api call response. Probably better to keep the name same as the response coming back from API

---
In reply to: [136439580](https://github.com/Microsoft/BotBuilder/pull/3324#discussion_r136439580) [](ancestors = 136439580)